kubernetes/resource_kubernetes_secret.go

```diff
@@ -119,17 +157,35 @@ func resourceKubernetesSecretUpdate(ctx context.Context, d *schema.ResourceData,
 	}
 
 	ops := patchMetadata("metadata.0.", "/metadata/", d)
-	if d.HasChange("data") {
-		oldV, newV := d.GetChange("data")
 
-		oldV = base64EncodeStringMap(oldV.(map[string]interface{}))
-		newV = base64EncodeStringMap(newV.(map[string]interface{}))
-
-		diffOps := diffStringMap("/data/", oldV.(map[string]interface{}), newV.(map[string]interface{}))
-
-		ops = append(ops, diffOps...)
+	newData := map[string]interface{}{}
```

Comment:
I've changed this to munge data and binary_data into a single map and just use the patch to overwrite the whole data field. This was easier than trying to work out each individual add/replace/delete operation.

Reply:
Nice! I like how readable this section is now.